lib/abci/handlers/initChainHandlerFactory.js

```diff
@@ -0,0 +1,35 @@
+const {
+  abci: {
+    ResponseInitChain,
+  },
+} = require('abci/types');
+
+/**
+ * Init Chain ABCI handler
+ *
+ * @param {BlockchainState} blockchainState
+ * @param {BaseLogger} logger
+ *
+ * @return {initChainHandler}
+ */
+function initChainHandlerFactory(
+  blockchainState,
+  logger,
+) {
+  /**
+   * @typedef endBlockHandler
+   *
+   * @param {abci.RequestInitChain} request
+   * @return {Promise<abci.ResponseInitChain>}
+   */
+  async function initChainHandler({ latestCoreChainLock }) {
+    logger.info('Init chain');
+    return new ResponseInitChain({
```

Comment:
It requires appHash https://docs.tendermint.com/master/spec/abci/abci.html#initchain

We should pass it from blockchainState.